internal/clients/kafka/topic/topic.go

```diff
@@ -0,0 +1,192 @@
+package topic
+
+import (
+	"context"
+	//"github.com/crossplane/crossplane-runtime/pkg/meta"
+	//"github.com/crossplane/crossplane-runtime/pkg/reconciler/managed"
+	//"github.com/twmb/franz-go/pkg/kerr"
+
+	"github.com/pkg/errors"
+	"github.com/twmb/franz-go/pkg/kadm"
+
+	"github.com/crossplane-contrib/provider-kafka/apis/topic/v1alpha1"
+)
+
+// Topic is a holistic representation of a Kafka Topic with all configurable
+// fields
+type Topic struct {
+	Name              string
+	ReplicationFactor int16
+	Partitions        int32
+	ID                string
+	Config            map[string]*string
+}
+
+// Get gets the topic from Kafka side and returns a Topic object.
+func Get(ctx context.Context, client *kadm.Client, name string) (*Topic, error) {
+
+	td, err := client.ListTopics(ctx, name)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot list topics")
+	}
+	if td[name].Err != nil {
+		return nil, errors.Wrap(td[name].Err, "topic does not exist in kafka cluster")
+	}
+
+	t, ok := td[name]
+	if !ok {
+		return nil, errors.New("no create response for topic")
+	}
+
+	tc, err := client.DescribeTopicConfigs(ctx, name)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot describe topics")
+	}
+
+	ts := Topic{}
+	ts.Name = name
+	ts.ReplicationFactor = int16(t.Partitions[0].Replicas[0])
+	ts.Partitions = t.Partitions[0].Partition
+	ts.ID = t.ID.String()
+	ts.Config = make(map[string]*string)
+
+	for _, value := range tc[0].Configs {
```

Comment:
Should we use [On](https://github.com/twmb/franz-go/blob/03c58cbf39e9cd53f457a71277b311fe4bcf4c66/pkg/kadm/configs.go#L60) method of ResourceConfigs? 

```suggestion
        rc, err := tc.On(name, nil)
        if err != nil {
                return nil, errors.Wrapf(err, "cannot find topic in describe result")
        }
        if rc.Err != nil {
		return nil, errors.Wrapf(rc.Err, "error in topic describe result")
	} 
        for _, value := range rc.Configs {
```

Not sure if we can safely use tc[0] here, @twmb what do you think?